go-controller/pkg/ovn/policy.go

```diff
@@ -1131,104 +951,50 @@ func (oc *Controller) deleteNetworkPolicy(policy *knet.NetworkPolicy) {
 	// Delete the port group
 	deletePortGroup(np.portGroupName)
 
-	// Go through each ingress rule.  For each ingress rule, delete the
-	// addressSet for the local peer pods.
-	for i := range np.ingressPolicies {
-		localPeerPods := fmt.Sprintf("%s.%s.%s.%d", policy.Namespace,
-			policy.Name, "ingress", i)
-		hashedAddressSet := hashedAddressSet(localPeerPods)
-		deleteAddressSet(hashedAddressSet)
-	}
-	// Go through each egress rule.  For each egress rule, delete the
-	// addressSet for the local peer pods.
-	for i := range np.egressPolicies {
-		localPeerPods := fmt.Sprintf("%s.%s.%s.%d", policy.Namespace,
-			policy.Name, "egress", i)
-		hashedAddressSet := hashedAddressSet(localPeerPods)
-		deleteAddressSet(hashedAddressSet)
+	// Delete ingress/egress address sets
+	for _, policy := range np.ingressPolicies {
+		policy.destroy()
+	}
+	for _, policy := range np.egressPolicies {
+		policy.destroy()
 	}
 }
 
 // handlePeerPodSelectorAddUpdate adds the IP address of a pod that has been
 // selected as a peer by a NetworkPolicy's ingress/egress section to that
 // ingress/egress address set
-func (oc *Controller) handlePeerPodSelectorAddUpdate(np *namespacePolicy,
-	addressMap map[string]bool, addressSet string, obj interface{}) {
-
+func (oc *Controller) handlePeerPodSelectorAddUpdate(as *addressSet, obj interface{}) {
 	pod := obj.(*kapi.Pod)
-	podAnnotation, err := util.UnmarshalPodAnnotation(pod.Annotations)
-	if err != nil {
-		return
-	}
-	// DUAL-STACK FIXME: handle multiple IPs
-	ipAddress := podAnnotation.IPs[0].IP.String()
-	if addressMap[ipAddress] {
-		return
-	}
-
-	np.Lock()
-	defer np.Unlock()
-	if np.deleted {
-		return
+	if err := as.AddPod(pod); err != nil {
+		klog.Warningf(err.Error())
```

Comment:
should this be an error instead of a warning?

Reply:
done